wrappers/services.go

```diff
@@ -481,6 +487,9 @@ func StopServices(apps []*snap.AppInfo, reason snap.ServiceStopReason, inter int
 		var err error
 		timings.Run(tm, "stop-service", fmt.Sprintf("stop service %q", app.ServiceName()), func(nested timings.Measurer) {
 			err = stopService(sysd, app, inter)
+			if err == nil && flags != nil && flags.Disable {
```

Comment:
I suggest to do an early check if flags == nil and give it an empty instance instead, so that it's not required to check for nil pointer at each step.